checkout/src/main/java/com/checkout/CheckoutApiServiceFactory.kt

```diff
@@ -31,6 +31,8 @@ import com.checkout.validation.validator.PhoneValidator
 import com.squareup.moshi.Moshi
 
 public object CheckoutApiServiceFactory {
+    private lateinit var correlationId: String
```

Comment:
Is the `correlationId` required to be cached? if not, shall we do it this way to avoid caching it in the factory?

```kt
    public fun create(
        publicKey: String,
        environment: Environment,
        context: Context,
    ): CheckoutApiService {
        val logger = EventLoggerProvider.provide()

        logger.setup(context, environment)

        return CheckoutApiClient(
            provideTokenRepository(context, publicKey, environment, logger.correlationId),
            provideThreeDSExecutor(logger),
        )
    }
```

And `provideTokenRepository` wii use the `correlationId` passed in